qiskit/circuit/library/standard_gates/pauli.py

```diff
@@ -0,0 +1,80 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Simulator command to perform multiple pauli gates in a single pass
+"""
+from qiskit.circuit.quantumregister import QuantumRegister
+from qiskit.circuit.library.standard_gates.i import IGate
+from qiskit.circuit.library.standard_gates.x import XGate
+from qiskit.circuit.library.standard_gates.y import YGate
+from qiskit.circuit.library.standard_gates.z import ZGate
+
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.exceptions import CircuitError
+
+
+class PauliGate(Gate):
+    r"""A multi-qubit Pauli gate.
+
+        This gate exists for optimization purposes for the
+        quantum statevector simulation, since applying multiple
+        pauli gates to different qubits at once can be done via
+        a single pass on the statevector.
+
+        The functionality is equivalent to applying
+        the pauli gates sequentially using standard Qiskit gates
+        """
+
+    def __init__(self, label="I"):
+        if isinstance(label, float):
+            label = "I"
+        self.pauli_string = label
+        super().__init__('pauli', len(label), [self.pauli_string[::-1]])
```

Comment:
Now that the pauli_string is the label shouldn't the parameter be `[self.pauli_string]` not the reversed string? It seems confusing to have two values stored and we should be consistent in using the same label order everywhere (in the class and in qobj).

Reply:
Fixed that; the only reverse being used in the code is now when constructing the `define` circuit.